### PR TITLE
REAL REMINDERS

### PR DESCRIPTION
**Pull request by jonwho**
remindme works for in-memory runs only

no persistent reminders if bot shuts down

### CODE REVIEW

**Comment by jonwho** [comment]
@BryanSLam merge my pr

**Comment by jonwho** [comment]
image

**Comment by BryanSLam** [comment]
`no persistent reminders if bot shuts down`
`no persistent reminders if bot shuts down`
`no persistent reminders if bot shuts down`
`no persistent reminders if bot shuts down`
`no persistent reminders if bot shuts down`

File: commands/reminder.go [hunk 1]
```diff
@@ -1,126 +1,83 @@
...
+	reminderCron.AddFunc(spec, func() {
+		s.ChannelMessageSend(m.ChannelID,
+			fmt.Sprintf("%s REMINDER! %s\n", m.Author.Mention(), reminderMessage))
+	})
+
+	s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("%s your reminder is set\n", m.Author.Mention()))
 }
 
-// TODO: figure out how to add logger here without creating a new discord session
+func cronSpec(value int, unit string) (string, error) {
```
Comment (BryanSLam) [inline commands/reminder.go]: are you lowercasing the time anywhere?

File: commands/reminder.go [hunk 2]
```diff
@@ -1,126 +1,83 @@
...
-//   // }
-//
-//   err := reminderClient.Add(message, date)
-//   if err != nil {
-//     s.ChannelMessageSend(m.ChannelID, err.Error())
-//     return
-//   }
-//   s.ChannelMessageSend(m.ChannelID, "Reminder Set!")
+	now := time.Now().Local()
+	switch unit {
```
Comment (BryanSLam) [inline commands/reminder.go]: hey wow this 12 case switch statement sure is better than a date, GEE WIZZ

File: commands/reminder.go [hunk 3]
```diff
@@ -1,126 +1,83 @@
...
+		spec = now.Add(time.Hour * 730 * time.Duration(value))
+	case "day":
+		spec = now.Add(time.Hour * 24 * time.Duration(value))
+	case "days":
+		spec = now.Add(time.Hour * 24 * time.Duration(value))
+	case "hour":
+		spec = now.Add(time.Hour * time.Duration(value))
+	case "hours":
+		spec = now.Add(time.Hour * time.Duration(value))
+	case "minute":
```
Comment (BryanSLam) [inline commands/reminder.go]: the use-case for having a reminder based on minutes is extremely low,

Comment (BryanSLam) [reply commands/reminder.go]: as in i don't think its needed

File: commands/reminder.go [hunk 4]
```diff
@@ -1,126 +1,83 @@
...
+		spec = now.Add(time.Minute * time.Duration(value))
+	case "minutes":
+		spec = now.Add(time.Minute * time.Duration(value))
+	default:
+		return "", errors.New("Unable to parse time for reminder")
+	}
+
+	seconds := spec.Second()
```
Comment (BryanSLam) [inline commands/reminder.go]: hey nice un-used variable lmao

**Review by BryanSLam — CHANGES_REQUESTED** [review]

**Comment by BryanSLam** [comment]
The reminder changes are bad because of the issues I listed ^^.  I'll merge the refactoring changes, but unless you show me the most impressive reminder function I've ever seen, I probably won't ever merge it because I'm partly pissed you worked on it without saying anything (we literally have 15 items in the backlog but you picked one that was already implemented charts???)

**Comment by jonwho** [comment]
ah shit i hear ya. dup work is a waste of time sorry for pretty much deleting all your work MY BAD

**Comment by BryanSLam** [comment]
yeah sorry for the long text, I was pretty annoyed when I woke up.  I think if we replace the cron calculation with something similar to the switch statement to calculating scheduling, but keep the redis as a way to store reminders, because deploying and crashes are going to happen and it'll wipe the reminders that are stored in-mem, then it could work

**Pull request closed without merging**